storm-client/src/jvm/org/apache/storm/daemon/Task.java

```diff
@@ -177,6 +196,35 @@ public BuiltinMetrics getBuiltInMetrics() {
         return builtInMetrics;
     }
 
+
+    // Non Blocking call. If cannot emmit to destination immediately, such tuples will be added to `pendingEmits` argument
+    public void sendUnanchored(String stream, List<Object> values, ExecutorTransfer transfer, Queue<AddressedTuple> pendingEmits) {
+        Tuple tuple = getTuple(stream, values);
+        List<Integer> tasks = getOutgoingTasks(stream, values);
+        for (Integer t : tasks) {
+            AddressedTuple addressedTuple = new AddressedTuple(t, tuple);
+            transfer.tryTransfer(addressedTuple, pendingEmits);
+        }
+    }
+
+    /**
+     * Send sampled data to the eventlogger if the global or component level debug flag is set (via nimbus api).
+     */
+    public void sendToEventLogger(Executor executor, List values,
```

Comment:
Will the method be called every emit when number of executors of event logger are more than 0? If then we may want to apply more optimizations here (not in this issue but other issue).